### DISCUSSION TITLE
Why dose AbstractClasspathClassLoader  need to load Class with Cache?

### DISCUSSION THREAD

**Discussion by smalljunHw** [sofastack/sofa-ark#687] [Q&A] [answered]
Since ClassLoader will only try to load class on the first time, it seems not neccessary to cache the Class Object.

**Comment by lvjing2** [comment, marked as answer]
正常情况是的，但是在一些找不到类的情况下，会出现不断查找问题。

**Reply by smalljunHw** [reply]
找不到类的话如果要再尝试加载，重新走一遍loadClass，感觉也是正常的，JDK的类加载设计如此，为什么在sofa ark的场景就需要特殊处理？

**Reply by lvjing2** [reply]
为了提高性能，不过这个不是必须的，如有需要可以提个需求，支持配置能力。
